### PR TITLE
Support packaging the core system as a Python package

### PR DESCRIPTION
**Pull request by bunjiboys**
* Moved migrations and templates to be within the package itself, included with `package_data`
* Use standard `setup.py` in favor of PBR based `setup.cfg` configuration
* Moved core requirements from requirements.txt to `setup.py` requirements
* Updated build scripts and documentation to reflect the new way of running the system

### CODE REVIEW

File: backend/cloud_inquisitor/utils.py [hunk 1]
```diff
@@ -133,10 +134,7 @@ def get_template(template):
     Returns:
         A Jinja2 Template object
     """
-    # Local import to prevent app startup failures
-    from cloud_inquisitor import app
-
-    tmplpath = os.path.join(app.config.get('BASE_PATH'), 'templates')
+    tmplpath = os.path.join(pkg_resources.resource_filename('cloud_inquisitor', 'data'), 'templates')
```
Comment (TheM0ng00se) [inline backend/cloud_inquisitor/utils.py]: while outside this commit may want to modify the e-mail recipient in the block above

**Comment by markofu** [comment]
Given these changes in how we build/install, surely we should reflect such in our install and general documentation?

**Comment by markofu** [comment]
This PR looks good to me. @HTTP500 can you review also?

**Comment by riot-gabe** [comment]
@markofu , looks to me like the install script was updated and I didn't see any other documentation involving `python manage.py`

**Comment by markofu** [comment]
@riot-gabe yep, yep - I chatted with @bunjiboys this afternoon and clarified my `documentation` question. Agree with your synopsis and on reflection, with Asbjorn, none of the `quickstart` or `install` documentation needs to be changed as the `install` process is not affected.

**Pull request merged**